src/cursor/cur_std.c

```diff
@@ -1290,6 +1290,60 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
     API_END_RET_STAT(session, ret, cursor_bound);
 }
 
+/*
+ * __wt_cursor_bounds_save --
+ *     Save cursor bounds to restore the state.
+ */
+int
+__wt_cursor_bounds_save(
+  WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *bounds_state)
+{
+    /* Save the bound flags to the state. */
+    bounds_state->bound_flags = F_MASK(cursor, WT_CURSTD_BOUND_ALL);
+
+    if (F_ISSET(cursor, WT_CURSTD_BOUND_LOWER)) {
+        WT_RET(__wt_scr_alloc(session, 0, &bounds_state->lower_bound));
```

Comment:
Consider `WT_RET(__wt_scr_alloc(session, cursor->lower_bound.size, &bounds_state->lower_bound));`, it may give better performance.  And same comment five lines down.

Reply:
[7ef3190](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/7ef3190b61449578ace49126ccf4c268948584fa)